src/components/ScanPatientPage/BarcodeScan.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+import Quagga from "quagga";
+import { useQuery } from "react-apollo";
+import { FETCH_ALL_PATIENTS } from "../../graphql/queries/patients";
+
+import "../../styles/BarcodeScan.css";
+
+const BarcodeScan = ({
+  eventID,
+  ccpID,
+}: {
+  eventID: string;
+  ccpID: string;
+}) => {
+  const history = useHistory();
+  const { data, loading } = useQuery(FETCH_ALL_PATIENTS);
+  const [barcode, setBarcode] = useState<string>("");
+
+  useEffect(() => {
+    if (!loading && barcode !== "") {
+      const selectedPatient = data.patients.filter(
+        (patient) => patient.barcodeValue.toString() === barcode
+      );
+
+      if (selectedPatient.length > 0) {
+        // Found patient
+        const {
+          collectionPointId: { id: patientCCPId },
+          id,
+        } = selectedPatient[0];
+        // Redirect to patient profile
+        history.replace(`/patients/edit/${patientCCPId}/${id}`);
+      } else {
+        // No existing patient
+        history.replace(`/patients/new/${ccpID}`);
+      }
+    }
+  }, [barcode]);
+
+  useEffect(() => {
+    Quagga.init(
+      {
+        inputStream: {
+          name: "Live",
+          type: "LiveStream",
+          target: document.querySelector("#barcode-scan"),
+          constraints: {
+            width: window.innerWidth,
+            height: window.innerHeight,
+            facing: "environment", // or user
+          },
+        },
+        decoder: {
+          readers: ["code_128_reader"],
+        },
+      },
+      function (err) {
+        if (err) {
+          console.log(err);
```

Comment:
Are we not doing error handling?

Reply:
this was boilerplate that I copied from Quagga's documentation, tbh I don't think we need this

Reply:
if we're gonna keep this console log here, you'll have to disable this line so the linter doesn't complain